### DISCUSSION TITLE
Don't validate all country areas

### DISCUSSION THREAD

**Discussion by mociepka** [saleor/saleor#11789] [Ideas]
### Context

[Google validations](https://github.com/mirumee/google-i18n-address) rules are great, but they need to work better with country areas.

In some countries like Italy, Russia, and China, data available for the country area is something other than what people expect.
In addition, address validation services like https://www.melissa.com/ don't validate contrite areas.

From experience, the best approach is to validate country areas only that contain official abbreviations like:
```
"US", "CA", "CN", "AU"
```

For other countries, validation introduces more issues with shipping than solutions.

### Proposition 1

Only validate counter areas that contain official abbreviations.

### Proposition 2

Add a settings interface to configure which rules you outline to use.

**Comment by szymi-** [comment]
Regardless of which of the two propositions above is accepted, it is important that it is implemented consistently across the entire Saleor API, wherever `AddressInput` is used and validated, e.g.: 

- `accountAddressCreate` mutation
- `accountAddressUpdate` mutation
- `addressCreate` mutation
- `addressUpdate` mutation
- `checkoutBillingAddressUpdate` mutation
- `checkoutCreate` mutation
- `checkoutShippingAddressUpdate` mutation
- `customerCreate` mutation
- `draftOrderCreate` mutation
- `orderUpdate` mutation
- `shopAddressUpdate` mutation
- `customerCreate` mutation
- `staffCreate` mutation
- `warehouseCreate` mutation
- `warehouseUpdate` mutation

and possibly more.

**Comment by mociepka** [comment]
RFC for this issue https://github.com/saleor/saleor/issues/12093
